api/v1alpha1/envoygateway_metrics_types.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// EnvoyGatewayMetrics defines control plane push/pull metrics configurations.
+type EnvoyGatewayMetrics struct {
+	// Address defines the address of Envoy Gateway Metrics Server.
+	Address *EnvoyGatewayMetricsAddress
```

Comment:
One last check point, I am not sure if we should serve the metrics server if we disabled prometheus.

If we do not serve the metrics server, we should be better to put `Address` into `EnvoyGatewayPrometheusProvider`.

If we still serve the metrics port if we disable the prometheus, the `Address` should be outside of the Prometheus.

cc @zirain @arkodg

Reply:
I would just remove Address for now, and hard code to `0.0.0.0:19001`, this is what is done in data plane

Reply:
It is Okay to me, but if people disabled Prometheus, should we still serve 0.0.0.0:19001?

Reply:
We shouldn't, that port should be disabled

Reply:
https://github.com/envoyproxy/gateway/pull/2021 is working on configuring metrics port, I think eg cp metrics port should be supported to be configurable.

Reply:
Yeah we should in the future, vote to add it in a future release than in this release. Getting the API right will require iterations and atm we don't have enough time to get it in

Reply:
Rmoved.